test/functional/feature_segwit.py

```diff
@@ -197,21 +191,21 @@ def run_test(self):
         assert_equal(self.nodes[1].getbalance(), 20 * Decimal("49.999"))
         assert_equal(self.nodes[2].getbalance(), 20 * Decimal("49.999"))
 
-        self.generate(self.nodes[0], 260)  # block 423
-
-        self.log.info("Verify witness txs are skipped for mining before the fork")
-        self.skip_mine(self.nodes[2], wit_ids[NODE_2][P2WPKH][0], True)  # block 424
-        self.skip_mine(self.nodes[2], wit_ids[NODE_2][P2WSH][0], True)  # block 425
-        self.skip_mine(self.nodes[2], p2sh_ids[NODE_2][P2WPKH][0], True)  # block 426
-        self.skip_mine(self.nodes[2], p2sh_ids[NODE_2][P2WSH][0], True)  # block 427
+        self.generate(self.nodes[0], 264)  # block 427
```

Comment:
Not really a fan of removing this test. Why not keep it to make the change in behaviour explicit and clarify that mining before segwit is active will result in a json exception?

Reply:
Done

Reply:
I disagree. Why do we need to test that a miner on a pre-segwit chain with segwit transactions in its mempool fails to mine a block? If we think that this scenario is possible (and so needs testing), then we shouldn't make the change in this PR, since failing to mine a block is strictly worse behaviour than mining a block without the segwit transaction.

Reply:
I guess this way we have test coverage for "unexpected witness"

Reply:
> I guess this way we have test coverage for "unexpected witness"

That's a very indirect way of testing a failure condition in `ContextualCheckBlock()`. If we want a test for that condition, then we should have a function test where a non-witness block (which is possible to create even after segwit activation) is stuffed with witness data. That's a condition that _could_ happen on mainnet, and so isn't relying on impossible-to-hit-in-production mining code. Even better would be a unit test for `ContextualCheckBlock()`.

I think it's inconsistent to remove production code that is "impossible to hit" and then add a test that hits that exact condition.

Reply:
Removing this test would also fix the "unexpected witness" CI failures